app/forms/schedule_form.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+class ScheduleForm
+  include ActiveModel::Model
+
+  attr_reader :errors, :flash_success
+
+  def initialize(params, user, schedule = nil)
+    @errors = ActiveModel::Errors.new self
+
+    # Not strong
+    @params = params
+    @current_user = user
+    @schedule = schedule
+  end
+
+  def save
+    editable_order_cycle_ids_for_create
+
+    return false if @params[:order_cycle_ids].blank?
+
+    @schedule.attributes = permitted_resource_params
+
+    if @schedule.save
+      @schedule.order_cycle_ids = @params[:order_cycle_ids]
+      @schedule.save!
+
+      sync_subscriptions_for_create
+
+    end
+
+    true
+  end
+
+  def update(_params)
```

Comment:
I put that there because I thought it was necessary for the update function to work since the `ResourceController` update takes a param. Should I just override the `ResourceController` function?

Reply:
I missed that, all good!